File: apps/desktop/src/services/NativeService/autostart.ts
import { invoke } from '@tauri-apps/api/core';

export const autostart = {
    isAutostartEnabled(): Promise<boolean> {
        return invoke<boolean>('is_autostart_enabled');
    },

    enableAutostart(): Promise<void> {
        return invoke('enable_autostart');
    },

    disableAutostart(): Promise<void> {
        return invoke('disable_autostart');
    },
} as const;

File: apps/desktop/src-tauri/src/commands/autostart.rs
// Copyright (c) 2026. 千诚. Licensed under GPL v3.

//! 自启动命令。
use tauri::{AppHandle, Runtime};

#[tauri::command]
pub fn enable_autostart<R: Runtime>(app: AppHandle<R>) -> Result<(), String> {
    crate::core::system::autostart::enable_autostart(app)
}

#[tauri::command]
pub fn disable_autostart<R: Runtime>(app: AppHandle<R>) -> Result<(), String> {
    crate::core::system::autostart::disable_autostart(app)
}

#[tauri::command]
pub fn is_autostart_enabled<R: Runtime>(app: AppHandle<R>) -> Result<bool, String> {
    crate::core::system::autostart::is_autostart_enabled(app)
}
